Kulyk_Lab/app/src/main/java/com/nazar/kulyk_lab/activities/MainActivity.java

```diff
@@ -43,35 +64,32 @@ private void getData() {
         Call<ArtList> call = rijksmuseumApi.getData();
 
         call.enqueue(new Callback<ArtList>() {
+            String TAG = getString(R.string.TAG);
             @Override
             public void onResponse(@NonNull Call<ArtList> call,
                                    @NonNull Response<ArtList> response) {
-                Log.d(TAG, "onResponse: ServerResponse: " + response.toString());
-
-                final ArtList responseBody = response.body();
-                if (responseBody != null){
-                    ArrayList<ArtObjects> artObjects = response.body().getArtObjects();
-                    displayData(artObjects);
-                } else {
-                    Log.e(TAG, getString(R.string.empty_response_body));
-                }
+                Log.d(TAG, response.toString());
+                ArrayList<ArtObjects> artObjects = Objects.requireNonNull(response.body()).getArtObjects();
+                displayData(artObjects);
             }
 
             @Override
             public void onFailure(@NonNull Call<ArtList> call, @NonNull Throwable t) {
-                Log.e(TAG, "onFailure: " + t.getMessage());
+                Log.e(TAG, t.getMessage());
+                no_data.setVisibility(View.VISIBLE);
             }
         });
     }
 
     private void displayData(ArrayList<ArtObjects> artObjects) {
-        for (int i = 0; i < artObjects.size(); i++) {
-            Log.d(TAG, artObjects.get(i).toString() +
-                    "\n----------------------------------------");
-        }
+        recyclerView.setLayoutManager(linerLayoutManager);
+        recyclerView.setAdapter(adapter);
+        adapter.addAll(artObjects);
```

Comment:
Setting layoutManager and adapter every time you reload data is not very convenient. You should set it just once and then just call `adapter.addAll` when data is loaded